.circleci/config.yml

```diff
@@ -0,0 +1,67 @@
+version: 2.1
+
+jobs:
+  build-and-test:
+    docker:
+      - image: cimg/python:3.11.4
+    steps:
+      - checkout
+
+      - run:
+          run: setup_creds
+          name: "Setup Credentials"
+          command: |
+            echo $BIGQUERY_SERVICE_ACCOUNT_JSON > ${HOME}/bigquery-service-key.json
+
+      - restore_cache:
+          key: &deps1-cache deps1-{{ .Branch }}-{{ checksum "requirements.txt" }}
+
+      - run:
+          name: "Setup dbt"
+          command: |
+            python3 -m venv venv
+            . venv/bin/activate
+
+            pip install --upgrade pip
+            pip install -r requirements.txt
+
+            mkdir -p ~/.dbt
+            cp integration_tests/ci/profiles.yml ~/.dbt/profiles.yml
+
+      - save_cache:
+          key: *deps1-cache
+          paths:
+            - "venv"
```

Comment:
Should we also cache `dbt_packages/` to speed up duplicate calls? We may also be able to call `dbt deps` in the `Setup dbt` step only, but I'm not entirely sure why it requires a `--target` option.

Reply:
Good suggestion! Had to write a little script to generate a composite checksum to capture the dbt dependencies, but it seems to work!